src/test/java/skeleton/RestSteps.java

```diff
@@ -60,12 +62,53 @@ public void i_add_users(List<User> users) {
         }
     }
 
+    @When("I delete user (.*?)$")
+    public void i_delete_user(String email) {
+        doRequest(
+                Unirest.delete(usersBaseUrl).queryString("email", email)
+        );
+
+    }
+
+    @When("I delete all users")
+    public void i_delete_all_users() {
+        doRequest(Unirest.delete(usersBaseUrl));
+    }
+
+    @When("I edit user (.*?) with (.*?) and (.*?)$")
+    public void i_edit_user(String email, String newFirstname, String newLastname) {
+
+        doRequest(
+                Unirest.post(usersBaseUrl + "/edit")
+                        .queryString("email", email)
+                        .body("{\"firstName\":\"" + newFirstname + "\",\"lastName\":\"" + newLastname + "\"}")
+                        .getHttpRequest()
+        );
+
+    }
+
     @Then("^there are (\\d+) users$")
     public void there_are_users(int expectedUsersCount) {
         doRequest(Unirest.get(usersBaseUrl));
         int actualUsersCount = response.getBody().getObject().getJSONArray("users").length();
         assertEquals("BBB!", expectedUsersCount, actualUsersCount);
     }
+
+    @Then("User (.*?) has firstname (.*?) and lastname (.*?)$")
+    public void user_has_credentials(String email, String firstname, String lastname) {
+        doRequest(Unirest.get(usersBaseUrl));
+        JSONArray users = response.getBody().getObject().getJSONArray("users");
+        int index;
+        for(index = 0; index < users.length(); index++) {
+            JSONObject userRecord = users.getJSONObject(index);
+            if (userRecord.getString("email").equals(email)) break;
+
+        }
+        String actualFirstname = users.getJSONObject(index).getString("first_name");
+        String actualLastname = users.getJSONObject(index).getString("last_name");
+
+        assertTrue(actualFirstname.equals(firstname) && actualLastname.equals(lastname));
```

Comment:
Solution 2 might look a little bit complicated but it's the best one when you have dozens of tests. I can show how code might actually look late if you are interesting. (Now I do not have IDE and have only HTTP access to github).